### DISCUSSION TITLE
Problem routing two levels in only one file

### DISCUSSION THREAD

**Discussion by bobarros** [vercel/next.js#20707] [Help] [answered]
Hi guys. First, I have to tell that I'm new at this world and usually lost with my code. Is very nice to find this space so active with so many people willing to help. Really =D

Let's jump in...

I'm bulding a website with four categories and my routes looks like this:

mywebsite.com.br/category/article-url

I tried to use Dynamic Routes two different ways.

**___________________ The way I want, but it's not working ___________________**

file: ./[url.js]

```
export async function getStaticPaths() {
  return {
      {
      params: {
        url: "category1/article1-url"
      }
      {
      params: {
        url: "category2/article2-url"
     }
    },
    fallback: false
  }
}
```

**___________________ The way I'm able to make it work ___________________**

file: ./category1/[url.js]
file: ./category2/[url.js]

And loading the right ulr list for each category in each url file:

```
export async function getStaticPaths() {
  return {
      {
      params: {
        url: "right-article-url-for-right-category"
      },
    fallback: false
  }
}
```
I hope I'm clear enough. 

Any suggestions how could I achieve the first option?

**Comment by Tomburgs** [comment, marked as answer]
There's few things.

1. Your object syntax is completely invalid.
2. `getStaticPaths` must return an array under `path` key with a list of urls.
3. `/[url].js` would catch only the first level of any url. If you want to catch urls which are deeper than that you must use a catch-all route (`[...url].js`). You can refer to [this documentation](https://nextjs.org/docs/api-routes/dynamic-api-routes#catch-all-api-routes).
4. Catch-all routes must return the path as an array, i.e if you want `category/article` you must return `["category", "article"]`.

So, it all together should look something like so:
`pages/[...url].js` ->
```javascript
{
  paths: [
    { params: { url: ["category", "article"] } }
    ...
  ],
  ...
}
```

You can refer to this documentation for any other [issues](https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation).

**Reply by bobarros** [reply]
This tip here was all I was really hoping for:

url: ["category", "article"] 

Thank you so much =D

**Reply by Tomburgs** [reply]
No problems. Feel free to set this as answered :)

**Reply by bobarros** [reply]
Sure thing. 
I took the time to test and is working just like I wanted when I asked my question. Thanks again.
